robot-name/robot_name.go

```diff
@@ -0,0 +1,83 @@
+// Manage robot factory settings.
+package robotname
+
+import (
+	"errors"
+	"fmt"
+	"math/rand"
+	"time"
+)
+
+// Robot has a name
+type Robot struct {
+	name string
+}
+
+// All the existing robot names so we can check for unique names
+var robots = make(map[string]Robot)
+
+// Name returns a robot's name in the format of two uppercase letters
+// followed by three digits. If the robot doesn't have a name already,
+// a random name is generated. Names must be random (no predictable pattern)
+// and unique (no duplicate names).
+func (r *Robot) Name() (string, error) {
+	if r.name == "" {
+		newName, error := GenerateUniqueName()
```

Comment:
Don't use `error` but `err`. That's the convention. Also, `error` is a type that you are hiding here.